### PR TITLE
Improve dashboard filter v2

### PR DESCRIPTION
**Pull request by Martchus**
Seems like I have to create a new PR. Here is how it looks like: http://imagebin.suse.de/2163/img

@okurz It should work now as discussed. One regex for everything, the "path" is separated by '/'.

### CODE REVIEW

File: lib/OpenQA/BuildResults.pm [hunk 1]
```diff
@@ -85,13 +85,29 @@ sub add_review_badge {
...
+                    push(@$group_ids, $child->id);
+                    push(@children,   $child);
+                }
+            }
+
+        }
+        else {
+            $group_ids = $group->child_group_ids;
+            @children  = $group->children;
+        }
```
Comment (okurz) [inline lib/OpenQA/BuildResults.pm]: IMHO getting too complicated. I would at least extract a method. You have a name for it already: "filter_subgroups" ;)

File: lib/OpenQA/Schema/Result/JobGroups.pm [hunk 2]
```diff
@@ -146,6 +146,22 @@ sub rendered_description {
...
+sub full_name {
+    my ($self) = @_;
+
+    if (my $parent = $self->parent) {
+        return $parent->name . ' / ' . $self->name;
+    }
+    else {
+        return $self->name;
+    }
+}
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/JobGroups.pm]: personal taste: How about

```perl
return $self->name unless $self->parent;
return $self->parent->name . ' / ' . $self->name;
```

**Comment by Martchus** [comment]
@okurz I fixed the mentioned issues. However, I discovered a new issue. Currently the parent-level statistics are filtered as well. As far as I remember you wanted those statistics to remain the same.

File: lib/OpenQA/Schema/Result/JobGroups.pm (hunk 2 above)
Comment (Martchus) [reply lib/OpenQA/Schema/Result/JobGroups.pm]: i changed it to

```
    if (my $parent = $self->parent) {
        return $parent->name . ' / ' . $self->name;
    }
    return $self->name;
```

This is short/readable enough and still saves querying the parent twice.

**Comment by okurz** [comment]
No, I think it's fine. If filtering then the parent statistics should also be filtered

**Comment by Martchus** [comment]
Ok, then I'll just drop the last commit again.

**Comment by Martchus** [comment]
Let's see whether re-triggering the CI helps as I can not reproduce this locally.

Note that the commit for preserving parent stats is still on https://github.com/Martchus/openQA/commit/c512a388975c469f2ee8e419b89572f2e762ce10 in case we want it after all. Making this behavior configurable should not be difficult either (but a thing for a new PR).

**Pull request merged**